mathesar/api/serializers/tables.py

```diff
@@ -103,6 +105,35 @@ def create(self, validated_data):
                 raise ProgrammingAPIException(e)
         return table
 
+    def update(self, instance, validated_data):
+        if self.partial:
+            # Save the fields that are stored in the model.
+            present_model_fields = []
+            for model_field in instance.MODEL_FIELDS:
+                if model_field in validated_data:
+                    setattr(instance, model_field, validated_data[model_field])
+                    present_model_fields.append(model_field)
+            instance.save(update_fields=present_model_fields)
+            for key in present_model_fields:
+                del validated_data[key]
+            # Save the fields that are stored in the underlying DB.
+            try:
+                instance.update_sa_table(validated_data)
+            except ValueError as e:
+                raise base_api_exceptions.ValueAPIException(e, status_code=status.HTTP_400_BAD_REQUEST)
+        return instance
+
+    def validate(self, data):
+        if self.partial:
+            columns = data.get('columns', None)
+            if columns is not None:
+                for col in columns:
+                    id = col.get('id', None)
+                    if id is None:
+                        message = "'id' field is required while batch updating columns."
+                        raise base_api_exceptions.NotFoundAPIException(ValidationError, message=message, status_code=status.HTTP_400_BAD_REQUEST)
```

Comment:
`NotFoundAPIException` suggests that a referenced resource does not exist, I'd say. In this case, the reference itself is incomplete. I'd say `MathesarValidationException` or a more specific subclass is preferable.

Reply:
That makes sense, I have refactored the above to raise a `MathesarValidationException`.